src/program.cpp

```diff
@@ -423,6 +423,12 @@ std::vector<argument> program::eval(parameter_map params) const
 #endif
 
     auto trace_level = value_of(MIGRAPHX_TRACE_EVAL{});
+    std::vector<argument> ret;
+
+    if(exec_env.async)
+    {
+        ctx.wait_for(exec_env.queue);
```

Comment:
Should this be asserting that `async` is true for this `exec_env` before calling `wait_for` ?

Reply:
No since we want to reuse code between eval() synchronous runs so if the user provides a stream but the flag isn't set, things will run normally as if there was no execution environment. 

Right now the default is to supply an execution environment with a false flag and null stream but, technically the null stream is valid in HIP and can potentially be passed around for other calls. I think the more concerning case is if we get to this call, and the user things they're achieving stream syncing, but not because something erroneously set the async flag to false.

 @pfultz2 , more as a sanity on this one,  is it worth to add an assert if we see  non-null stream and false in the execution env? I've just assumed we'll always have things set correctly via the API/default assignment from eval() signature otherwise.